crates/sui-types/src/temporary_store.rs

```diff
@@ -541,22 +541,13 @@ impl<S> TemporaryStore<S> {
             self.reset(gas, gas_status);
         }
 
-        if let Err(err) = self
-            .charge_gas_for_storage_changes(gas_status, gas_object_id)
-            .and_then(|total_bytes_written_deleted| {
-                gas_status.charge_computation_gas_for_storage_mutation(total_bytes_written_deleted)
```

Comment:
Now that this is gone, I think we should update the doc comment for the function, since the flow is now a lot simpler. I also might add a `debug_assert` at the end of the function to check that we don't change the computation gas after 534 (since a lot of the problems we had before had to deal with sneaky computation gas charges intermingled with charging for storage)